dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java

```diff
@@ -474,7 +475,8 @@ public Period createPeriod( DateInterval dateInterval )
         final DateTimeUnit from = cal.toIso( dateInterval.getFrom() );
         final DateTimeUnit to = cal.toIso( dateInterval.getTo() );
 
-        return new Period( this, from.toJdkDate(), to.toJdkDate(), getIsoDate( from ) );
+        return new Period( this, from.toJdkDate(), to.toJdkDate(), cal.name() == "persian" ? getIsoDate( dateInterval.getFrom() ) : getIsoDate( from ));
```

Comment:
Please remove the hard-coding of this specific calendar in PeriodType. We would like PeriodType to be unaware of any period types. This should be built into the Calendar solution.